Counting/main_counting_cc.yag

```diff
@@ -82,15 +82,16 @@
 
 		{{/*Countingrole part*/}}
 		{{$sdict := sdict}}{{with (dbGet 0 "countingroleblock")}}{{$sdict = .Value}}{{end}}
-		{{if and $countingrole ($sdict.Get (str .User.ID)|not)}}
-			{{if not $blockrole}}
-				{{addRoleID $countingrole}}
-				{{if ne $db.countingrole .User.ID}}
-					{{takeRoleID $db.countingrole $countingrole}}
-				{{end}}
-				{{$db.Set "countingrole" .User.ID}}
-			{{end}}
+		{{if $countingrole}}
+            		{{if and (ne $db.countingrole .User.ID) (targetHasRoleID $db.countingrole $countingrole)}}
+                		{{takeRoleID $db.countingrole $countingrole}}
+            		{{end}}
```

Comment:
Need to wrap this in a try-catch due to `targetHasRoleID` requiring the target user be a member of the server

```go
{{try}}
            		{{if and (ne $db.countingrole .User.ID) (targetHasRoleID $db.countingrole $countingrole)}}
                		{{takeRoleID $db.countingrole $countingrole}}
            		{{end}}
					{{catch}}
						{{/* Baka Left*/}}
					{{end}}```

Reply:
changed